src/AutoBalance.cpp

```diff
@@ -247,6 +249,22 @@ class AutoBalance_PlayerScript : public PlayerScript
             if (mapABInfo->mapLevel < player->getLevel())
                 mapABInfo->mapLevel = player->getLevel();
         }
+
+        void OnGiveXP(Player* player, uint32& amount, Unit* victim) override
+        {
+            if (victim && FullGroupXP)
+            {
+                Map* map = player->GetMap();
+
+                if (map->IsDungeon())
+                {
+                    // Ensure that the players always get the same XP, even when entering the dungeon alone
+                    uint32 maxPlayerCount = ((InstanceMap*)sMapMgr->FindMap(map->GetId(), map->GetInstanceId()))->GetMaxPlayers();
+                    uint32 currentPlayerCount = map->GetPlayersCountExceptGMs();
```

Comment:
so when we test it won't work with a GM account? or only if the GM mode is activated?

Reply:
The function "GetPlayersCountExceptGMs" only counts players, not GMs with enabled GM mode. Each GM who disables GM mode is treated as a normal player.